src/helpers/getPlaystreakArgsFromQuestData.ts

```diff
@@ -8,7 +8,7 @@ export function getPlaystreakArgsFromQuestData(
   questMeta: Quest,
   questPlayStreakData: UserPlayStreak | undefined | null,
   useModuleInitTimeForSessionStartTime?: boolean
-): PlayStreakEligibility {
+): Omit<PlayStreakEligibility, 'onSync'> {
```

Comment:
I'm purposely omitting this because I don't think we should return the sync handler here. We'd have to include all the props from `UseSyncPlayStreakProps`

Reply:
This is the implementation on the client:

```tsx
const { syncPlayStreak } = useSyncPlayStreak({
    refreshPlayStreak: questPlayStreakResult.invalidateQuery
})

<QuestDetails
  ...
    playStreak: {
    ...getPlaystreakArgsFromQuestData(questMeta, questPlayStreakData),
        onSync: async () => syncPlayStreak(selectedQuestId)
    }
/>
```

Reply:
Why not add `useSyncPlayStreak` directly to the quest details wrapper component above? This way the re-renders are contained to just the wrapper component.

This fits the existing pattern like for:

```
  useSyncPlaySession(
    projectId,
    questPlayStreakResult.invalidateQuery,
    syncPlaySession
  )
```

where we pass in the syncPlaySession function as a prop. (which is window.api on client and a fetch elsewhere).

so you would just take in 

```
syncPlayStreakWithExternalSource: (params: {
    quest_id: number
    signature: string
  }) => Promise<unknown>
```

as a quest details wrapper prop and pass it to `useSyncPlayStreak`

Reply:
you also wouldn't need to pass in all the props in this case. just the mutation fxn call returned from `useSyncPlayStreak`

Reply:
My main target was to prevent having to include another required prop to the QuestRewardWrapper props because this is technically a prop that won't be used unless the quest type is play streak.

The play streak prop is optional and only used for play streak quests, so that was my main reason for choosing this place.

On second thought, it's fine to add it. I don't love its API, but it's fine, this is a pretty niche and complex implementation that won't be used elsewhere.